src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/GoldPan.java

```diff
@@ -165,13 +168,34 @@ public void updateRandomizer() {
         List<ItemStack> recipes = new LinkedList<>();
 
         for (GoldPanDrop drop : drops) {
-            if (drop.getValue() > 0) {
-                recipes.add(new ItemStack(getInputMaterial()));
+            if (drop.getValue() <= 0) {
+                continue;
+            }
+
+            for (Material material : getInputMaterials()) {
+                recipes.add(new ItemStack(material));
                 recipes.add(drop.getOutput());
             }
         }
 
         return recipes;
     }
 
+    /**
+     * This is a helper method to determine whether the input item was a valid item.
+     *
+     * @param material
+     *              The {@link Material} that was input.
+     *
+     * @return whether or not the provided material is allowed.
+     */
+    public @Nonnull boolean isCorrectInputMaterial(Material material) {
+        for (Material m : getInputMaterials()) {
+            if (SlimefunUtils.isItemSimilar(new ItemStack(material), new ItemStack(m), true, false)) {
```

Comment:
WHAT!?!?!?!?!?

So instead of just checking if two Materials are the same (m == material) you wrap them as ItemStacks and literally call the most expensive method in the entire project to do a simple `==` comparison...
Like... WTF

Reply:
its something I left in after refactoring, ill clean it up